WebAssembly/Runtime/OpCodeCompilationException.cs

```diff
@@ -16,9 +16,26 @@ public OpCodeCompilationException(OpCode opCode, string message)
             this.OpCode = opCode;
         }
 
+        /// <summary>
+        /// Creates a new <see cref="OpCodeCompilationException"/> with the provided parameters.
+        /// </summary>
+        /// <param name="miscellaneousOpCode">The miscellaneous operation attempted.</param>
+        /// <param name="message">An explanation of the problem, concatenated with <paramref name="miscellaneousOpCode"/> and passed to the base as the message.</param>
+        public OpCodeCompilationException(MiscellaneousOpCode miscellaneousOpCode, string message)
+            : base($"{miscellaneousOpCode} {message}")
+        {
+            this.OpCode = OpCode.MiscellaneousOperationPrefix;
+            this.MiscellaneousOpCode = miscellaneousOpCode;
+        }
+
         /// <summary>
         /// The operation attempted.
         /// </summary>
         public OpCode OpCode { get; }
+
+        /// <summary>
+        /// The miscellaneous operation attempted, if <see cref="OpCode"/> is <see cref="OpCode.MiscellaneousOperationPrefix"/>.
+        /// </summary>
+        public MiscellaneousOpCode? MiscellaneousOpCode { get; }
```

Comment:
I feel like this could get out of hand as more opcode pairs are introduced, I doubt that this will be the last we'll see in the coming years.
Maybe instead inherit `OpCodeCompilationException` from a new `MiscellaneousOpCodeCompilationException`, adding `MiscellaneousOpCode ` as non-nullable there?
Except that won't work with things like `StackTooSmallException`, hmm... I need to think on this...

Reply:
Indeed, I originally did add `MiscellaneousOpCodeCompilationException`, but then I realised I'd need to add new classes for each exception type. And that seemed worse than adding additional properties in the existing exception types. If you can think of a better way to do it I'd be happy to change it.